cli/src/commands/jobs.rs

```diff
@@ -188,13 +190,61 @@ pub async fn handle_submission(api: &mut PhylumApi, matches: &clap::ArgMatches)
     }
 
     if synch {
+        if pretty_print {
+            if let Err(err) = vulnreach(api, matches, packages, job_id.to_string()).await {
+                print_user_failure!("Reachability analysis failed: {err:?}");
+            }
+        }
+
         log::debug!("Requesting status...");
         print_job_status(api, &job_id, ignored_packages, pretty_print).await
     } else {
         Ok(ExitCode::Ok)
     }
 }
 
+/// Perform vulnerability reachability analysis.
+async fn vulnreach(
+    api: &mut PhylumApi,
+    matches: &clap::ArgMatches,
+    packages: Vec<PackageDescriptor>,
+    job_id: String,
+) -> Result<()> {
+    // Skip requests early if we know user doesn't have the required role.
+    if !api.roles().contains(&RealmRole::Vulnreach) {
+        return Ok(());
```

Comment:
This is an undocumented pre-requisite. Perhaps a `debug` or `warn` log message would help here?

Local testing was unable to proceed past this point. The early return path was taken for both STAGING and PRODUCTION, both before and after a `phylum auth login` (to update/refresh the refresh token). To be certain, the code was modified to provide a bit more output:

```
    // Skip requests early if we know user doesn't have the required role.
    log::warn!("Checking for `Vulnreach` role in refresh token ...");
    if !api.roles().contains(&RealmRole::Vulnreach) {
        log::warn!("NO `Vulnreach` role in refresh token");
        return Ok(());
    }
    log::warn!("`Vulnreach` role found in refresh token");
```

...and the result was always the same:

```
../phylum-ui  6  18 on  develop [✘?] via 🐳 desktop-linux via  v20.2.0 took 4s
❌ 100 ❯ ~/dev/phylum/localdev/cli/target/debug/phylum analyze -l reacher_test -p delme_ui -v
✅ Successfully parsed lockfile "./yarn.lock" as type: yarn
✅ Job ID: 41ef3aeb-3e5e-4b6e-83d0-ee2f4edf983a
[2023-06-01T01:37:24Z WARN  phylum_cli::commands::jobs] Checking for `Vulnreach` role in refresh token ...
[2023-06-01T01:37:24Z WARN  phylum_cli::commands::jobs] NO `Vulnreach` role in refresh token
# Phylum OSS Supply Chain Risk Analysis - FAILED

This repository analyzes the risk of new dependencies. An
administrator of this repository has set requirements via Phylum policy.

If you see this comment, one or more dependencies have failed Phylum's risk analysis.

### Package: `is-typed-array@1.1.10` failed.

#### is-typed-array@1.1.10 may be a typosquatted package

- **Risk Domain**: Malicious Code
- **Risk Level**: high
- **Reason**: risk level cannot exceed medium

### Package: `yaml@2.1.3` failed.

#### yaml@2.1.3 is vulnerable to Uncaught Exception

- **Risk Domain**: Software Vulnerability
- **Risk Level**: high
- **Reason**: risk level cannot exceed medium

[View this project in the Phylum UI](https://app.phylum.io/projects/42abb122-d6c3-4305-b9e1-3cfe00be87dc?label=reacher_test)
```

(That is PROD, but STG was the same). Are there other requirements for getting vulnreach output? Is there a recommended testing methodology to confirm the behavior in this PR?

Reply:
A debug log might make sense, but this should definitely not warn. Access to the vuln reach API is controlled by the role, so skipping this code when the user doesn't have the role is expected behavior.

Reply:
Yeah I'd like to avoid any output here since this is basically our form of "feature detection" and we'd always shove pointless warnings into people's faces if we would log this.